### PR TITLE
Create Products with Redis Done

### LINKED ISSUE #147: Refactor Create Products for Redis Integration
Refactor the code to work with a Redis Database

### PR DESCRIPTION
**Pull request by Pratik-Karnik**
Closes #147

### CODE REVIEW

File: app/models.py [hunk 1]
```diff
@@ -19,27 +19,38 @@ def __init__(self, uid=0, sid=0,subtotal=0.0,products=[]):
         self.subtotal = float(subtotal)
         self.products = products
 
-    def save(self):
-        if self.sid == 0:
-            self.sid = self.__next_index()
+    def save(self,type):
```
Comment (jiweix) [inline app/models.py]: why add this type parameter here?

Comment (Pratik-Karnik) [reply app/models.py]: Bcoz when we save products, we dont need to again increment the index for sid. For this differentiation, I added that.

Comment (jiweix) [reply app/models.py]: isn't the sid == 0 enough? for existing shopcart, this sid should not be 0

Comment (Pratik-Karnik) [reply app/models.py]: Ya I think you are right!! Pushed new changes now!

**Comment by jiweix** [comment]
have you run the test? I think the create products tests are all failing.

**Comment by Pratik-Karnik** [comment]
@jiweix I havent refactored the TDD code yet

**Comment by Pratik-Karnik** [comment]
There is another epic for TDD refactoring

**Comment by jiweix** [comment]
The TDD code should not need be refactored. This is the whole point of writing tests.

**Pull request merged**